### PR TITLE
Refactor metrics tests for reuse

### PR DESCRIPTION
**Pull request by edgarRd**
This PR includes the following changes:
- Refactor Metrics tests for reuse.
- Relocate truncation tests that are not parquet specific.

@rdblue @vgankidi

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -231,6 +231,19 @@ project(':iceberg-core') {
       exclude group: 'org.slf4j', module: 'slf4j-log4j12'
     }
   }
+
+  task testJar(type: Jar){
+    archiveClassifier = 'tests'
+    from sourceSets.test.output
+  }
+
+  configurations {
```
Comment (rdsr) [inline build.gradle]: We've define `testArtifacts` configuration in `iceberg-hive` as well. I think it makes sense to define in under `allprojects` or `subprojects` once.  Same for `testJar`

File: core/src/test/java/org/apache/iceberg/TestMetricsTruncation.java [hunk 2]
```diff
@@ -17,45 +17,44 @@
...
 
-public class TestParquetMetricsTruncation {
+public class TestMetricsTruncation {
   @Test
-  public void testTruncateBinaryMin() throws IOException {
+  public void testTruncateBinaryMin() {
     ByteBuffer test1 = ByteBuffer.wrap(new byte[] {1, 1, (byte) 0xFF, 2});
     // Output of test1 when truncated to 2 bytes
-    ByteBuffer test1_2_expected = ByteBuffer.wrap(new byte[] {1, 1});
+    ByteBuffer test12Expected = ByteBuffer.wrap(new byte[] {1, 1});
```
Comment (rdblue) [inline core/src/test/java/org/apache/iceberg/TestMetricsTruncation.java]: I think the underscores make the test more readable. Keeping them would also remove a lot of non-functional changes from this PR, so I think it is a good idea to revert the name style changes.

**Comment by rdblue** [comment]
@edgarRd, nice work! Just one thing to fix and I'll merge this.

File: core/src/test/java/org/apache/iceberg/TestMetricsTruncation.java (hunk 2 above)
Comment (edgarRd) [reply core/src/test/java/org/apache/iceberg/TestMetricsTruncation.java]: I agree, but I'd need to disable the `checkstyle` rule for this.

Comment (rdblue) [reply core/src/test/java/org/apache/iceberg/TestMetricsTruncation.java]: Can you add `@SuppressWarnings("checkstyle:(rulename)")` to the class?

Comment (edgarRd) [reply core/src/test/java/org/apache/iceberg/TestMetricsTruncation.java]: Just added. Thanks!

**Comment by edgarRd** [comment]
It seems like there are some non-deterministic errors in the test suite already in master:
```
org.apache.iceberg.spark.source.TestIcebergSourceHiveTables > testHiveFilesTable FAILED
    java.lang.RuntimeException at TestIcebergSourceHiveTables.java:196
        Caused by: org.apache.thrift.transport.TTransportException at TestIcebergSourceHiveTables.java:196
            Caused by: java.net.SocketException at TestIcebergSourceHiveTables.java:196

org.apache.iceberg.spark.source.TestIcebergSourceHiveTables > testHiveFilesUnpartitionedTable FAILED
    java.lang.RuntimeException at TestIcebergSourceHiveTables.java:233
        Caused by: org.apache.thrift.transport.TTransportException at TestIcebergSourceHiveTables.java:233
            Caused by: java.net.SocketException at TestIcebergSourceHiveTables.java:233

org.apache.iceberg.spark.source.TestIcebergSourceHiveTables > testHiveManifestsTable FAILED
    java.lang.RuntimeException at TestIcebergSourceHiveTables.java:438
        Caused by: org.apache.thrift.transport.TTransportException at TestIcebergSourceHiveTables.java:438
            Caused by: java.net.SocketException at TestIcebergSourceHiveTables.java:438

org.apache.iceberg.spark.source.TestIcebergSourceHiveTables > testHiveHistoryTable FAILED
    java.lang.RuntimeException at TestIcebergSourceHiveTables.java:285
        Caused by: org.apache.thrift.transport.TTransportException at TestIcebergSourceHiveTables.java:285
            Caused by: java.net.SocketException at TestIcebergSourceHiveTables.java:285

org.apache.iceberg.spark.source.TestIcebergSourceHiveTables > testHiveEntriesTable FAILED
    java.lang.RuntimeException at TestIcebergSourceHiveTables.java:144
        Caused by: org.apache.thrift.transport.TTransportException at TestIcebergSourceHiveTables.java:144
            Caused by: java.net.SocketException at TestIcebergSourceHiveTables.java:144
```

**Comment by rdblue** [comment]
I restarted CI. I'll have to take a look at the flaky test, too.

**Comment by rdblue** [comment]
Merged. Thanks for fixing this, @edgarRd!

**Comment by edgarRd** [comment]
Great, thanks!

**Pull request merged**